### PR TITLE
Only publish each Pulp repository once

### PR DESCRIPTION
**Pull request by twaugh**
Signed-off-by: Tim Waugh <twaugh@redhat.com>

### CODE REVIEW

**Comment by twaugh** [comment]
@mlangsdorf

File: atomic_reactor/plugins/exit_pulp_publish.py [hunk 1]
```diff
@@ -81,8 +81,9 @@ def publish_to_crane(self, repo_prefix="redhat-"):
         self.pulp_handler.create_dockpulp()
         if not repo_prefix:
             repo_prefix = ''
-        pulp_repos = ['%s%s' % (repo_prefix, image.pulp_repo) for image in image_names]
-        self.pulp_handler.publish(pulp_repos)
+        pulp_repos = set(['%s%s' % (repo_prefix, image.pulp_repo)
```
Comment (lcarva) [inline atomic_reactor/plugins/exit_pulp_publish.py]: Out of curiosity, when will this list contain more than one item?

File: atomic_reactor/plugins/exit_pulp_publish.py [hunk 2]
```diff
@@ -81,8 +81,9 @@ def publish_to_crane(self, repo_prefix="redhat-"):
...
+                          for image in image_names])
+        self.pulp_handler.publish(sorted(list(pulp_repos)))
```
Comment (vrutkovs) [inline atomic_reactor/plugins/exit_pulp_publish.py]: Not sure if `sorted` and `list` are required - dockpulp's `crane` seems to accept any iterable

Comment (twaugh) [inline atomic_reactor/plugins/exit_pulp_publish.py]: Right you are. Updated.

File: atomic_reactor/plugins/exit_pulp_publish.py (hunk 1 above)
Comment (twaugh) [inline atomic_reactor/plugins/exit_pulp_publish.py]: It iterates through the image names, so there is an `image.pulp_repo` for each tag we made.

Comment (lcarva) [inline atomic_reactor/plugins/exit_pulp_publish.py]: Sure, but isn't the pulp_repo name derived from the `name` label in Dockerfile?
`pulp_repo` does not include the registry, nor the tag.

**Comment by twaugh** [comment]
No release note.

File: atomic_reactor/plugins/exit_pulp_publish.py (hunk 1 above)
Comment (twaugh) [inline atomic_reactor/plugins/exit_pulp_publish.py]: It is. So yes, we expect the resulting set to have a single element.

**Pull request merged**